core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java

```diff
@@ -96,30 +102,34 @@ public Class getNotificationTypeClass() {
     public NotificationCenter() {
         notificationsListeners.put(NotificationType.Activate, new ArrayList<NotificationHolder>());
         notificationsListeners.put(NotificationType.Track, new ArrayList<NotificationHolder>());
-        notificationsListeners.put(NotificationType.Decision, new ArrayList<NotificationHolder>());
     }
 
     // private list of notification by notification type.
     // we used a list so that notification order can mean something.
     private Map<NotificationType, ArrayList<NotificationHolder>> notificationsListeners = new HashMap<NotificationType, ArrayList<NotificationHolder>>();
+    private ArrayList<NotificationHolder> decisionListenerHolder = new ArrayList<>();
 
     /**
      * Convenience method to support lambdas as callbacks in later version of Java (8+).
      *
-     * @param decisionNotificationListenerInterface
+     * @param decisionNotificationListener
      * @return greater than zero if added.
      */
-    public int addDecisionNotificationListener(final DecisionNotificationListenerInterface decisionNotificationListenerInterface) {
-        if (decisionNotificationListenerInterface instanceof DecisionNotificationListener) {
-            return addNotificationListener(NotificationType.Decision, (NotificationListener) decisionNotificationListenerInterface);
-        } else {
-            return addNotificationListener(NotificationType.Decision, new DecisionNotificationListener() {
-                @Override
-                public void onDecision(@Nonnull String type, @Nonnull String userId, @Nonnull Map<String, ?> attributes, @Nonnull Map<String, ?> decisionInfo) {
-                    decisionNotificationListenerInterface.onDecision(type, userId, attributes, decisionInfo);
+    public int addDecisionNotificationListener(DecisionNotificationListener decisionNotificationListener) {
+        int id = this.notificationListenerID.incrementAndGet();
```

Comment:
Consider moving this call to where `id` is actually needed. Since there are early-returns in this method, we could bump the id sequence unnecessarily.